psamm/datasource/sbml.py

```diff
@@ -711,11 +712,14 @@ def write_model(self, file, model):
         reaction_genes = {}
         reaction_equations = {}
         reaction_names = {}
+        biomass_id = ''
         for r in model.parse_reactions():
             if (model_reactions is not None and
                     r.id not in model_reactions):
                 continue
-
+            if r.id == model.get_biomass_reaction():
+                biomass_id = self._create_unique_id(
+                    reaction_equations, self._make_safe_id(r.id))
```

Comment:
You are generating a new ID so I don't think this will work correctly. The objective has to reference the same ID as you are using for the reaction.